heft-plugins/heft-sass-plugin/src/SassPlugin.ts

```diff
@@ -3,88 +3,101 @@
-          await this._runSassTypingsGeneratorAsync(
-            heftSession,
-            heftConfiguration,
-            build.properties.watchMode
-          );
-        });
-      });
+  public apply(taskSession: IHeftTaskSession, heftConfiguration: HeftConfiguration): void {
+    taskSession.hooks.clean.tapPromise(PLUGIN_NAME, async (cleanOptions: IHeftTaskCleanHookOptions) => {
+      const sassConfiguration: ISassConfiguration = await this._loadSassConfigurationAsync(
+        heftConfiguration,
+        taskSession.logger
+      );
+      cleanOptions.addDeleteOperations({ sourcePath: sassConfiguration.generatedTsFolder });
+      for (const secondaryGeneratedTsFolder of sassConfiguration.secondaryGeneratedTsFolders || []) {
+        cleanOptions.addDeleteOperations({ sourcePath: secondaryGeneratedTsFolder });
+      }
+      for (const cssOutputFolder of sassConfiguration.cssOutputFolders || []) {
+        cleanOptions.addDeleteOperations({ sourcePath: cssOutputFolder });
+      }
+    });
+
+    taskSession.hooks.run.tapPromise(PLUGIN_NAME, async (runOptions: IHeftTaskRunHookOptions) => {
+      // TODO: Handle watch mode
+      await this._runSassTypingsGeneratorAsync(taskSession, heftConfiguration, false);
     });
   }
 
   private async _runSassTypingsGeneratorAsync(
-    heftSession: HeftSession,
+    taskSession: IHeftTaskSession,
     heftConfiguration: HeftConfiguration,
     isWatchMode: boolean
   ): Promise<void> {
-    const logger: ScopedLogger = heftSession.requestScopedLogger('sass-typings-generator');
     const sassConfiguration: ISassConfiguration = await this._loadSassConfigurationAsync(
       heftConfiguration,
-      logger
+      taskSession.logger
     );
     const sassTypingsGenerator: SassTypingsGenerator = new SassTypingsGenerator({
-      buildFolder: heftConfiguration.buildFolder,
       sassConfiguration
     });
 
     await sassTypingsGenerator.generateTypingsAsync();
     if (isWatchMode) {
-      Async.runWatcherWithErrorHandling(async () => await sassTypingsGenerator.runWatcherAsync(), logger);
+      Async.runWatcherWithErrorHandling(
+        async () => await sassTypingsGenerator.runWatcherAsync(),
+        taskSession.logger
+      );
     }
   }
 
   private async _loadSassConfigurationAsync(
     heftConfiguration: HeftConfiguration,
-    logger: ScopedLogger
+    logger: IScopedLogger
   ): Promise<ISassConfiguration> {
     const { buildFolder } = heftConfiguration;
     const sassConfigurationJson: ISassConfigurationJson | undefined =
-      await SassTypingsPlugin._getSassConfigurationLoader().tryLoadConfigurationFileForProjectAsync(
+      await SassPlugin._getSassConfigurationLoader().tryLoadConfigurationFileForProjectAsync(
         logger.terminal,
         buildFolder,
         heftConfiguration.rigConfig
       );
 
+    // Set defaults if no configuration file or option was found
     return {
-      ...sassConfigurationJson
+      ...sassConfigurationJson,
```

Comment:
Stick this at the end and you can get rid of all of the `sassConfigurationJson?.propName ||`s

Reply:
Good call.